addons/web/static/src/js/views/view_dialogs.js

```diff
@@ -113,6 +115,7 @@ var FormViewDialog = ViewDialog.extend({
         this.shouldSaveLocally = options.shouldSaveLocally;
         this.readonly = options.readonly;
         this.deletable = options.deletable;
```

Comment:
maybe you could have this.deletable = options.deletable && this.showRemove; and then, not using showRemove elsewhere in this file